Update script.sh: add checks

commitay opened the pull request:
Refs: https://github.com/caskroom/homebrew-cask/pull/41111

Add checks for leftover apps, launchjobs and pkgs.

Edit: added kexts (may as well check for them while we can) and both launchjob scripts.

I've dropped the checks in `HOME` and `/Library` because of false positives.

commitay commented:
@vitorgalvao @reitermarkus 

Any ideas as to what I'm doing wrong with this? The output always stops (and the build fails) after the first iteration of the loop.

https://travis-ci.org/caskroom/homebrew-cask/builds/309312282#L757
```
>>> Leftover: apps
> /Applications/TeamViewer.app
> /Applications/TeamViewer.app/Contents
```

commitay commented:
Actually works now. https://travis-ci.org/caskroom/homebrew-cask/builds/314175643#L352

```
>>> Leftover: apps
> /Applications/TeamViewer.app
>>> Leftover: launchjob_install
> com.teamviewer.Helper
> com.teamviewer.desktop
> com.teamviewer.service
> com.teamviewer.teamviewer
>>> Leftover: launchjob_load
> com.teamviewer.Helper
>>> Leftover: pkgs
> com.teamviewer.teamviewer13Agent
> com.teamviewer.teamviewer13AuthPlugin
> com.teamviewer.teamviewer13Restarter
> com.teamviewer.teamviewer13
> com.teamviewer.teamviewer13Font
> com.teamviewer.teamviewer13PriviledgedHelper

The command ". ci/travis/script.sh" exited with 1.
```

commitay commented:
This has ended up being more complicated than originally planned.

Added `brew bundle` dump/cleanup so `depends_on cask:` doesn't cause false positives.

Moved `brew tap` to `before_script.sh` so the output of `script.sh` is cleaner.

```
Running script.sh...

==> Tapping homebrew/bundle
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-bundle'...
remote: Counting objects: 2128, done.
remote: Compressing objects: 100% (22/22), done.
remote: Total 2128 (delta 12), reused 14 (delta 6), pack-reused 2100
Receiving objects: 100% (2128/2128), 371.28 KiB | 7.28 MiB/s, done.
Resolving deltas: 100% (1210/1210), done.
Tapped 0 formulae (72 files, 562.5KB)

>>> No Casks modified, skipping
The command ". ci/travis/script.sh" exited with 0.

```
```
Running script.sh...

>>> No Casks modified, skipping
The command ". ci/travis/script.sh" exited with 0.
```

reitermarkus commented on ci/travis/before_script.sh:
```diff
@@ -13,8 +13,8 @@ header 'Running before_script.sh...'
 
-# update homebrew
-run brew update
+# this step also updates homebrew
```
I‘d just keep the `brew update` command and remove this comment, just to be explicit.

commitay commented:
Might merge this if everyone is okay with it, I will monitor PRs for any issues.

commitay commented:
Changed this up again, moved brew tap / brew bundle inside the install conditional. Only 54 Casks with `depends_on cask:` in Caskroom so it's rather unnecessary to run tap/bundle for every build.
